build/build.go

```diff
@@ -595,6 +595,28 @@ func toSolveOpt(ctx context.Context, node builder.Node, multiDriver bool, opt Op
 		so.FrontendAttrs["attest:provenance"] = "mode=min,inline-only=true"
 	}
 
+	var prvmode string
+	if v, ok := opt.Attests["attest:provenance"]; ok {
+		if prvmode, err = getProvenanceMode(*v); err != nil {
+			return nil, nil, err
+		}
+	}
+	gitprv, err := getGitProvenance(ctx, prvmode, opt.Inputs.ContextPath, opt.Inputs.DockerfilePath)
+	if err != nil {
+		return nil, nil, err
+	}
+	if gitprv.Source != "" {
+		so.FrontendAttrs["label:"+specs.AnnotationSource] = gitprv.Source
+		so.FrontendAttrs["vcs:source"] = gitprv.Source
+	}
+	if gitprv.Revision != "" {
+		so.FrontendAttrs["label:"+specs.AnnotationRevision] = gitprv.Revision
+		so.FrontendAttrs["vcs:revision"] = gitprv.Revision
+	}
+	if gitprv.DockerfilePath != "" {
+		so.FrontendAttrs["label:"+DockerfileLabel] = gitprv.DockerfilePath
+	}
```

Comment:
Keep the current behavior that sets git provenance labels introduced in https://github.com/docker/buildx/pull/1297.